runner/operator/cmo_ch/v2_1_0/qc_agg/nucleo_qc_operator_agg.py

```diff
@@ -0,0 +1,189 @@
+import os
+import json
+import uuid
+import logging
+from pathlib import Path
+from jinja2 import Template
+from beagle import settings
+from runner.operator.operator import Operator
+from runner.models import RunStatus, Port, Run
+from runner.run.objects.run_creator_object import RunCreator
+from file_system.models import File, FileGroup, FileType
+
+
+logger = logging.getLogger(__name__)
+
+WORKDIR = os.path.dirname(os.path.abspath(__file__))
+
+
+meta_fields = [
+    "igoId",
+    settings.CMO_SAMPLE_TAG_METADATA_KEY,
+    settings.CMO_SAMPLE_NAME_METADATA_KEY,
+    settings.CMO_SAMPLE_CLASS_METADATA_KEY,
+    settings.PATIENT_ID_METADATA_KEY,
+    "investigatorSampleId",
+    settings.ONCOTREE_METADATA_KEY,
+    "tumorOrNormal",
+    "tissueLocation",
+    settings.SAMPLE_CLASS_METADATA_KEY,
+    "sampleOrigin",
+    "preservation",
+    "collectionYear",
+    "sex",
+    "species",
+    "tubeId",
+    "cfDNA2dBarcode",
+    "baitSet",
+    "qcReports",
+    "barcodeId",
+    "barcodeIndex",
+    settings.LIBRARY_ID_METADATA_KEY,
+    "libraryVolume",
+    "libraryConcentrationNgul",
+    "dnaInputNg",
+    "captureConcentrationNm",
+    "captureInputNg",
+    "captureName",
+]
+
+
+class CMOCHNucleoOperatorQcAgg(Operator):
+    """
+    Operator for the ACCESS QC workflow:
+
+    https://github.com/msk-access/access_qc_generation/blob/master/access_qc.cwl
+
+    This Operator will search for Nucleo Bam files based on an IGO Request ID
+    """
+
+    def get_jobs(self):
+
+        sample_inputs = self.get_nucleo_outputs()
+
+        return [
+            RunCreator(
+                **{
+                    "name": "CMO-CH Nucleo QC: %s, %i of %i" % (self.request_id, i + 1, len(sample_inputs)),
+                    "app": self.get_pipeline_id(),
+                    "inputs": job,
+                    "tags": {settings.REQUEST_ID_METADATA_KEY: self.request_id, "cmoSampleId": job["sample_name"]},
+                }
+            )
+            for i, job in enumerate(sample_inputs)
+        ]
+
+    def get_nucleo_outputs(self):
```

Comment:
Could we modify this method in the similar way we did here: https://github.com/mskcc/beagle/pull/1116 (ie adding the case when runs are passed in the chaining process)

Reply:
Just added the fix!